pkg/backend/httpstate/backend.go

```diff
@@ -1129,7 +1131,64 @@ func (b *cloudBackend) Preview(ctx context.Context, stack backend.Stack,
 func (b *cloudBackend) Update(ctx context.Context, stack backend.Stack,
 	op backend.UpdateOperation,
 ) (sdkDisplay.ResourceChanges, error) {
-	return backend.PreviewThenPromptThenExecute(ctx, apitype.UpdateUpdate, stack, op, b.apply)
+	return backend.PreviewThenPromptThenExecute(ctx, apitype.UpdateUpdate, stack, op, b.apply, b)
+}
+
+func (b *cloudBackend) isCopilotFeatureEnabled(projectName tokens.PackageName, opts display.Options) bool {
+	// Have copilot features been requested by specifying the --copilot flag to the cli
+	if !opts.ShowCopilotFeatures {
+		return false
+	}
+
+	// Is copilot enabled this project in Pulumi Cloud
+	enabled, ok := b.copilotEnabledForProject[projectName]
+	contract.Assertf(ok,
+		"copilotEnabledForProject has not been set for project %q. only available after an update has been started.",
+		projectName)
+	return enabled
+}
+
+// explain takes engine events, renders them out to a buffer as something similar to what the user sees
+// in the CLI, and then explains the output with Copilot.
+func (b *cloudBackend) explain(
```

Comment:
What is the value of having these private helper methods separate from the public Explain/IsEnabled? They seem to have exactly the same signatures?

Reply:
To decouple the internal names which reflect what the functions actually do, vs. the interface name which reflects how it is used.

- In the case of `explain` its the same
- but as you note for `IsEnabledForProject` we have:
   - how we use the fn in generic apply context `IsEnabledForProject`
   - the implementation of that in the Pulumi Cloud backend is actually `isCopilotFeatureEnabled`

This whole abstraction does feel a bit funny. Maybe passing in extra options to the interactive selection would feel a bit cleaner?..

Reply:
However we don't implementing the "Stack" interface directly via the Backend instance, and instead have another lightweight type that wraps backend. I will follow that pattern here too and add an "Explainer" struct instead.